src/data_tools/data_transformer.py

```diff
@@ -0,0 +1,867 @@
+        ret[c,d] = [x,y,z]
+
+    return ret
+
+def apply_mosaic(image, seed, swaps):
+    '''
+    Randomly swaps sections of an image. Sections are rectangles of random dimensions
+    
+    INPUTS
+    ------
+    image: Input image of dimensions WIDTH x HEIGHT.
+    seed: Allows you to repreduce results. If called with the same seed twice, the output will be the same.
+        any integer
+    swaps: determines the number of times pixels are swapped.
+        any integer
+
+    NOTES
+    -----
+    Must be applied to labels as well.
+    '''
+    ret = np.copy(image)
+    cpy = np.copy(image)
+
+    for k in range(swaps):
+        random.seed(seed*k)
+        box_height = random.randint(50,200)
+        random.seed(seed*k + 50)
+        box_width = random.randint(50,200)
+        random.seed(seed*k + 100)
+        a = random.randint(0, HEIGHT - box_height - 1)
+        random.seed(seed*k + 200)
+        b = random.randint(0, WIDTH - box_width - 1)
+        random.seed(seed*k + 300)
+        c = random.randint(0, HEIGHT - box_height - 1)
+        random.seed(seed*k + 200)
+        d = random.randint(0, WIDTH - box_width - 1)
+        random.seed(seed*k + 300)
+        
+        cpy[a:(a + box_height), b:(b + box_width)] = ret[a:(a + box_height), b:(b + box_width)]
+        ret[a:(a + box_height), b:(b + box_width)] = ret[c:(c + box_height), d:(d + box_width)]
+        ret[c:(c + box_height), d:(d + box_width)] = cpy[a:(a + box_height), b:(b + box_width)]
+
+    return ret
+
+def uniform_mosaic(image, seed, v_box, h_box):
+    '''
+    Splits image into a grid of x_box by y_box equal rectangles. Randomly swaps rectangles.
+    
+    INPUTS
+    ------
+    image: Input image of dimensions WIDTH x HEIGHT.
+    seed: Allows you to repreduce results. If called with the same seed twice, the output will be the same.
+        any integer
+    v_box: determines the number of boxes along the vertical axis.
+        any positive integer less than 721
+    h_box: determines the number of boxes along the horizontal axis.
+        any positive integer less than 1281
+
+    NOTES
+    -----
+    Must be applied to labels as well.
+    '''
+    arr = np.arange(1, v_box * h_box + 1, dtype=int) # middle index is not swapped if odd
+    np.random.seed(seed)
+    np.random.shuffle(arr)
+
+    rows = int(HEIGHT/v_box)
+    cols = int(WIDTH/h_box)
+
+    ret = np.copy(image)
+    cpy = np.copy(image)
+
+    for num in range(int((v_box*h_box)/2)):
+        c1 = ((arr[2*num] - 1) % h_box) * cols
+        r1 = int((arr[2*num]-1)/h_box) * rows
+        c2 = ((arr[2*num + 1] - 1) % h_box) * cols
+        r2 = int((arr[2*num + 1]-1)/h_box) * rows
+        
+        cpy[r1:(r1 + rows), c1:(c1 + cols)] = ret[r1:(r1 + rows), c1:(c1 + cols)]
+        ret[r1:(r1 + rows), c1:(c1 + cols)] = ret[r2:(r2 + rows), c2:(c2 + cols)]
+        ret[r2:(r2 + rows), c2:(c2 + cols)] = cpy[r1:(r1 + rows), c1:(c1 + cols)]
+
+    return ret
+
+def apply_saltpepper(image, seed, density=10):
```

Comment:
Maybe instead of just black, white, and grey, potentially add just random colours? Just food for thought. This can be a later addition.